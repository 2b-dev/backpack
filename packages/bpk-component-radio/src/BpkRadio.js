import React, { PropTypes } from 'react';

import './bpk-radio.scss';

const BpkRadio = (props) => {
  const classNames = ['bpk-radio'];
  const { name, label, disabled, className, ...rest } = props;

  if (disabled) { classNames.push('bpk-radio--disabled'); }
  if (className) { classNames.push(className); }

  // This is awkward because the label-has-for rule enforces an 'id' / 'for' pairing
  // when it's not really necessary for nested inputs.
  // See https://github.com/evcohen/eslint-plugin-jsx-a11y/issues/51.
  /* eslint-disable jsx-a11y/label-has-for */
  return (
    <label className={classNames.join(' ')}>
      <input
        type="radio"
        className="bpk-radio__input"
        name={name}
        disabled={disabled}
        {...rest}
      />
      &nbsp;
      {label}
    </label>
  );
  /* eslint-enable */
};

BpkRadio.propTypes = {
  name: PropTypes.string.isRequired,
  label: PropTypes.string.isRequired,
  disabled: PropTypes.bool,
  className: PropTypes.string,
};

BpkRadio.defaultProps = {
  disabled: false,
  className: null,
};

export default BpkRadio;
