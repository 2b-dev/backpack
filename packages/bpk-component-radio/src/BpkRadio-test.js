import React from 'react';
import renderer from 'react/lib/ReactTestRenderer';
import BpkRadio from './BpkRadio';

describe('BpkRadio', () => {
  it('should render correctly', () => {
    const tree = renderer.create(
      <BpkRadio
        name="radio"
        onChange={() => null}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with id attribute', () => {
    const tree = renderer.create(
      <BpkRadio
        id="radio"
        name="radio"
        onChange={() => null}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with checked attribute', () => {
    const tree = renderer.create(
      <BpkRadio
        name="radio"
        onChange={() => null}
        checked
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with label attribute', () => {
    const tree = renderer.create(
      <BpkRadio
        name="radio"
        onChange={() => null}
        label="My label"
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with disabled attribute', () => {
    const tree = renderer.create(
      <BpkRadio
        name="radio"
        onChange={() => null}
        disabled
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render correctly with value attribute', () => {
    const tree = renderer.create(
      <BpkRadio
        name="radio"
        value="my-value"
        onChange={() => null}
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

