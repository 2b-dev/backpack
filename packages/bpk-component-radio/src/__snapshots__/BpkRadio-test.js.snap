exports[`BpkRadio should render correctly 1`] = `
<label
  className="bpk-radio">
  <input
    checked={false}
    className="bpk-radio__input"
    disabled={false}
    id={null}
    name="radio"
    onChange={[Function onChange]}
    type="radio"
    value="radio" />
</label>
`;

exports[`BpkRadio should render correctly with checked attribute 1`] = `
<label
  className="bpk-radio">
  <input
    checked={true}
    className="bpk-radio__input"
    disabled={false}
    id={null}
    name="radio"
    onChange={[Function onChange]}
    type="radio"
    value="radio" />
</label>
`;

exports[`BpkRadio should render correctly with disabled attribute 1`] = `
<label
  className="bpk-radio bpk-radio--disabled">
  <input
    checked={false}
    className="bpk-radio__input"
    disabled={true}
    id={null}
    name="radio"
    onChange={[Function onChange]}
    type="radio"
    value="radio" />
</label>
`;

exports[`BpkRadio should render correctly with id attribute 1`] = `
<label
  className="bpk-radio">
  <input
    checked={false}
    className="bpk-radio__input"
    disabled={false}
    id="radio"
    name="radio"
    onChange={[Function onChange]}
    type="radio"
    value="radio" />
</label>
`;

exports[`BpkRadio should render correctly with label attribute 1`] = `
<label
  className="bpk-radio">
  <input
    checked={false}
    className="bpk-radio__input"
    disabled={false}
    id={null}
    name="radio"
    onChange={[Function onChange]}
    type="radio"
    value="radio" />
   My label
</label>
`;

exports[`BpkRadio should render correctly with value attribute 1`] = `
<label
  className="bpk-radio">
  <input
    checked={false}
    className="bpk-radio__input"
    disabled={false}
    id={null}
    name="radio"
    onChange={[Function onChange]}
    type="radio"
    value="my-value" />
</label>
`;
